driver/razerkbd_driver.c

```diff
@@ -1714,8 +1730,13 @@ static int razer_raw_event(struct hid_device *hdev, struct hid_report *report, u
         return 0;
     }
 
+    // FN can't be present
+    if (size != 16 && usb_dev->descriptor.idProduct != USB_DEVICE_ID_RAZER_BLACKWIDOW_2019) {
+        return 0;
+    }
+
     // The event were looking for is 16 bytes long and starts with 0x04
-    if(intf->cur_altsetting->desc.bInterfaceProtocol == USB_INTERFACE_PROTOCOL_KEYBOARD && size == 16 && data[0] == 0x04) {
+    if(intf->cur_altsetting->desc.bInterfaceProtocol == USB_INTERFACE_PROTOCOL_KEYBOARD && data[0] == 0x04) {
```

Comment:
Why was this change done (plus the few lines above)?

Reply:
Hi,

I've observed size=22 when Fn key is pressed on this keyboard.
So, the changes requested will break Fn key again.

Let me know if I can help with some more debug

Reply:
Yes please do so.

Reply:
Done.
Regards,

Reply:
as owner of a blackwidow 2019 I disagree about "fn key is a minor issue" ;p

But off course I can understand the workflow!